library/core/helpers/class.ssoutils.php

```diff
@@ -37,12 +37,13 @@ class SsoUtils {
     /** @var string */
     private $stateToken;
 
+    /** @var Logger  */
     private $logger;
 
     /**
      * SsoUtils constructor.
      */
-    public function __construct(Gdn_Configuration $config, Cookie $cookie, Gdn_Session $session, Logger $logger) {
+    public function __construct(Gdn_Configuration $config, Cookie $cookie, Gdn_Session $session, ?\Psr\Log\LoggerInterface $logger = null) {
```

Comment:
You're still going to want to grab this out of the container, manually, inside the constructor if `$logger` is `null`. Otherwise, you could potentially have it not be set and your calls to its methods further down will fatal.

Reply:
@initvector That is not necessary in the latest versions of Vanilla. At least not since `garden-container` 3.0: https://github.com/vanilla/garden-container/releases/tag/v3.0.0

> Add support for null defaults for classes that don't exist.

Reply:
> That is not necessary in the latest versions of Vanilla. At least not since garden-container 3.0

I'd mentioned in an earlier comment, but my concern wasn't instantiation through the container, but rather instantiation _without_ the container. This class is a couple years old now. I don't know what kinda usage there is in the wild. If there's anyone manually instantiating this thing, it's likely to break if we aren't compensating for this new dependency not automatically being injected.